lib/protein/adapters/amqp_adapter/server.ex

```diff
@@ -91,8 +90,22 @@ defmodule Protein.AMQPAdapter.Server do
   defp init_conn_chan_queue(url, queue) do
     case Connection.open(url) do
       {:ok, conn} ->
-        {:ok, chan} = Channel.open(conn)
-        Queue.declare(chan, queue)
+        chan =
+          try do
+            {:ok, chan} = Channel.open(conn)
+            {:ok, _} = Queue.declare(chan, queue, durable: true)
+            Logger.info("Declared queue #{queue} as durable")
+            chan
+          catch
+            :exit, reason = {{:shutdown, {:server_initiated_close, 406, _message}}, _} ->
+              IO.inspect(reason)
```

Comment:
We shouldn't do this in production code.

Reply:
I have found no other way to do it, if you have any ideas feel free to share them.

Reply:
No other way to do what?

Reply:
Shit, sorry for the confusion - somehow I thought that this comment relates to the condition of `catch`. In regards to the actual line, I agree, changing it to `Logger.debug` as @bartkekbsh suggested seem ok with you?

Reply:
Yup, `Logger.debug` is fine.